src/Microsoft.Framework.ConfigurationModel.Xml/XmlConfigurationExtension.cs

```diff
@@ -7,7 +10,25 @@ public static class XmlConfigurationExtension
     {
         public static IConfigurationSourceRoot AddXmlFile(this IConfigurationSourceRoot configuration, string path)
         {
-            configuration.Add(new XmlConfigurationSource(path));
+            return AddXmlFile(configuration, path, optional: false);
+        }
+
+        public static IConfigurationSourceRoot AddXmlFile(this IConfigurationSourceRoot configuration, string path, bool optional)
+        {
+            if (string.IsNullOrEmpty(path))
```

Comment:
I am under the impression that [NotNull] won't check for empty string. I guess this is the reason the check wasn't done that way. Do you still want me to change that?